application/org.openjdk.jmc.flightrecorder.flameview/src/main/java/org/openjdk/jmc/flightrecorder/flameview/views/FlameGraphView.java

```diff
@@ -300,14 +299,23 @@ public Void call() throws Exception {
 			view.modelState = ModelState.CALCULATION;
 			StacktraceModel model = new StacktraceModel(view.threadRootAtTop, view.frameSeparator, items);
 			TraceNode root = TraceTreeUtils.createRootWithDescription(items, model.getRootFork().getBranchCount());
+			if (isInvalid) {
+				return abort();
+			}
 			TraceNode traceNode = TraceTreeUtils.createTree(root, model);
-			String jsonModel = view.toJSonModel(traceNode).toString();
+			String jsonModel = view.toJSonModel(traceNode, isInvalid).toString();
```

Comment:
Java is pass by value. You'll need to pass a reference to this (the ModelRebuildCallable) and query that for isInvalid(), otherwise you'll only have the value at the time of the invocation passed around.

Reply:
Correct! all object references are passed by value... should wait till morning to not overlook it.

Reply:
corrected!